hooks/useTikTokAnaylsis.tsx

```diff
@@ -92,25 +104,36 @@ const useTikTokAnalysis = () => {
       }
       setSettingMode(SETTING_MODE.CREATE);
       setThought(STEP_OF_ANALYSIS.CREATING_ARTIST);
-      const artistInfo = await saveSetting(
-        profileWithComments.nickname,
-        profileWithComments.avatar,
-        SETTING_MODE.CREATE,
-      );
-      setSelectedArtist({ ...artistInfo });
-      setArtistActive(true);
-
+      let artistId = "";
+      while (1) {
+        const artistInfo = await saveSetting(
+          profileWithComments.nickname,
+          profileWithComments.avatar,
+          SETTING_MODE.CREATE,
+        );
+        if (artistInfo) {
+          setSelectedArtist({ ...artistInfo });
+          artistId = artistInfo?.id;
+          setArtistActive(true);
+          break;
+        }
+      }
       setThought(STEP_OF_ANALYSIS.SAVING_ANALYSIS);
-      const data = await saveAnalysis({
-        ...profileWithComments,
-        segments: [...fanSegmentsWithIcons],
-        chat_id: newId,
-        artistId: artistInfo.id,
-      });
-      setResult({
-        ...profileWithComments,
-        id: data.id,
-      });
+      while (1) {
+        const data = await saveAnalysis({
+          ...profileWithComments,
+          segments: [...fanSegmentsWithIcons],
+          chat_id: newId,
+          artistId,
+        });
+        if (data) {
+          setResult({
+            ...profileWithComments,
+            id: data.id,
+          });
+          break;
+        }
+      }
       setThought(STEP_OF_ANALYSIS.FINISHED);
     } catch (error) {
```

Comment:
this hook is too long

Developer Requirements Document
```
### **Code Standards**

- Maximum file length: 99 lines
- Directory Structure:
    - `/components`: HTML code only
    - `/libs`: Helper functions and API calls
    - `/hooks`: State management and helper function calls
    - `/providers`: Hooks with state shared across multiple components

### **Coding Guidelines**

- Follow principles from "Clean Code" by Robert C. Martin (Uncle Bob)
- Single Responsibility Principle: Each file should perform precisely one function
```
https://zany-hourglass-531.notion.site/Developer-Requirements-Document-15026a7d835c8062bbcde49b59aa00e8?pvs=74